parsers/s01-parse/mstilkerich/bind9-logs.yaml

```diff
@@ -0,0 +1,39 @@
+---
+filter: "Upper(evt.Parsed.program) == 'NAMED'"
+debug: true
+onsuccess: next_stage
+pattern_syntax:
+  BIND9_TIMESTAMP_DATE: '%{MONTHDAY:ts_d}-%{MONTH:ts_m}-%{YEAR:ts_y}'
+  BIND9_TIMESTAMP_TIMEMS: '\d{3}'
+  BIND9_TIMESTAMP_TIME: '%{TIME:ts_t}\.%{BIND9_TIMESTAMP_TIMEMS:ts_ms}'
+  BIND9_TIMESTAMP: '%{BIND9_TIMESTAMP_DATE} %{BIND9_TIMESTAMP_TIME}'
+  # categories: security, client, update-security, etc.
+  BIND9_CATEGORY: '[a-z]+'
+  BIND9_SEVERITY: '[a-z]+'
+  BIND9_PREFIX: '^(%{BIND9_TIMESTAMP:timestamp} )?(%{BIND9_CATEGORY}: )?(%{BIND9_SEVERITY}: )?client @0x[0-9a-fA-F]+ %{IP:remote_addr}#%{POSINT:remote_port}.*: '
+name: mstilkerich/bind9
+description: "Parse bind9 logs"
+nodes:
+  # When multiple grok patterns are specified, only the first matching one is applied
+  - grok:
+      pattern: '^%{BIND9_PREFIX}zone transfer .* (?:denied|\(NOTAUTH\))\s*$'
+      apply_on: message
+  - grok:
+      pattern: '^%{BIND9_PREFIX}bad zone transfer request: ''\S+/IN'': non-authoritative zone (?:denied|\(NOTAUTH\))\s*$'
+      apply_on: message
+  - grok:
+      pattern: '^%{BIND9_PREFIX}(?:view (?:internal|external): )?query(?: \(cache\))? .* (?:denied|\(NOTAUTH\))\s*$'
+      apply_on: message
+statics:
+  - meta: service
+    value: bind9
+  - meta: log_type
+    value: bind9_denied
```

Comment:
This is okay at the moment, however, if we support multiple patterns they might need to be applied under each GROK pattern

Reply:
I don't understand what you mean by supporting multiple patterns (there already are multiple patterns in the parser), but I also take you don't currently see an issue here. Anyway I saw no reason to tag the different policy violation event types differently and therefore all are tagged as `bind9_denied` type.